Add a Loader malloc heap zone

rreno opened the pull request:
#### 66b6285e8f367770bb1acc584f8ff1f270259b26
<pre>
Add a Loader malloc heap zone
<a href="https://bugs.webkit.org/show_bug.cgi?id=260623">https://bugs.webkit.org/show_bug.cgi?id=260623</a>
rdar://114340888

Reviewed by Brent Fulgham and Simon Fraser.

This adds a generic Loader allocator that fast malloc classes can use in
the loading subsystem. This will allow us to categorize classes with
WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER into a loader malloc zone
without the need to make a zone for each class.

Use by including &quot;LoaderMalloc.h&quot; (if necessary) and replacing
WTF_MAKE_FAST_ALLOCATED with
WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Loader) in your loader class.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/loader/ApplicationManifestLoader.h:
* Source/WebCore/loader/ContentFilter.h:
* Source/WebCore/loader/CrossOriginPreflightResultCache.h:
* Source/WebCore/loader/DocumentThreadableLoader.h:
* Source/WebCore/loader/FrameLoader.cpp:
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/HistoryController.h:
* Source/WebCore/loader/ImageLoader.h:
* Source/WebCore/loader/LinkPreloadResourceClients.h:
* Source/WebCore/loader/LoaderMalloc.cpp: Copied from Source/WebCore/loader/ProgressTrackerClient.h.
* Source/WebCore/loader/LoaderMalloc.h: Copied from Source/WebCore/loader/ProgressTrackerClient.h.
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebCore/loader/NavigationDisabler.h:
* Source/WebCore/loader/NavigationScheduler.cpp:
* Source/WebCore/loader/NavigationScheduler.h:
* Source/WebCore/loader/PolicyChecker.h:
* Source/WebCore/loader/ProgressTracker.cpp:
* Source/WebCore/loader/ProgressTracker.h:
* Source/WebCore/loader/ProgressTrackerClient.h:
* Source/WebCore/loader/ResourceLoadObserver.h:
* Source/WebCore/loader/ResourceTiming.h:
* Source/WebCore/loader/SubframeLoader.h:
* Source/WebCore/loader/SubresourceLoader.h:
* Source/WebCore/loader/TextTrackLoader.h:
* Source/WebCore/loader/ThreadableLoaderClient.h:
* Source/WebCore/loader/WorkerThreadableLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.h:
* Source/WebCore/loader/appcache/ApplicationCacheHost.h:
* Source/WebCore/loader/archive/ArchiveResourceCollection.h:
* Source/WebCore/loader/cache/CachedFontLoadRequest.h:
* Source/WebCore/loader/cache/CachedResource.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/loader/cache/CachedResourceRequestInitiatorTypes.h:
* Source/WebCore/loader/cache/CachedSVGDocumentReference.h:
* Source/WebCore/loader/cache/MemoryCache.h:
* Source/WebCore/loader/icon/IconLoader.h:
* Source/WebCore/loader/ios/LegacyPreviewLoader.h:

Canonical link: <a href="https://commits.webkit.org/267205@main">https://commits.webkit.org/267205@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a6e724c37ae4174895a947a9ff3237cbd40f1a71

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15956 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16275 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16662 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17719 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14980 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16142 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19285 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16376 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17437 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16147 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16624 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13607 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18479 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13868 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14424 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/21296 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14858 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14590 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17838 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15183 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/12874 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14423 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3811 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18791 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15005 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/loader/ApplicationManifestLoader.h:
```diff
@@ -40,7 +41,7 @@ class CachedApplicationManifest;
 
 class ApplicationManifestLoader final : private CachedRawResourceClient {
-WTF_MAKE_NONCOPYABLE(ApplicationManifestLoader); WTF_MAKE_FAST_ALLOCATED;
+WTF_MAKE_NONCOPYABLE(ApplicationManifestLoader); WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Loader);
```
Nit: This should be two lines.

brentfulgham replied:
This style is also the cause of the `style` bot failures.

rreno replied:
Yeah as you suspected I just kept the style that was there before.

rreno replied:
Interestingly it's not the only line like this but the style checker only caught one. And it caught a different one on my local machine!

brentfulgham reviewed (approved):
I'm not a fan of the multiple macros on a single line, but that's the existing style, so I guess just keep it.

r=me.

rreno committed:
Add a Loader malloc heap zone
https://bugs.webkit.org/show_bug.cgi?id=260623
rdar://114340888

Reviewed by Brent Fulgham and Simon Fraser.

This adds a generic Loader allocator that fast malloc classes can use in
the loading subsystem. This will allow us to categorize classes with
WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER into a loader malloc zone
without the need to make a zone for each class.

Use by including "LoaderMalloc.h" (if necessary) and replacing
WTF_MAKE_FAST_ALLOCATED with
WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Loader) in your loader class.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/loader/ApplicationManifestLoader.h:
* Source/WebCore/loader/ContentFilter.h:
* Source/WebCore/loader/CrossOriginPreflightResultCache.h:
* Source/WebCore/loader/DocumentThreadableLoader.h:
* Source/WebCore/loader/FrameLoader.cpp:
* Source/WebCore/loader/FrameLoader.h:
* Source/WebCore/loader/HistoryController.h:
* Source/WebCore/loader/ImageLoader.h:
* Source/WebCore/loader/LinkPreloadResourceClients.h:
* Source/WebCore/loader/LoaderMalloc.cpp: Copied from Source/WebCore/loader/ProgressTrackerClient.h.
* Source/WebCore/loader/LoaderMalloc.h: Copied from Source/WebCore/loader/ProgressTrackerClient.h.
* Source/WebCore/loader/LocalFrameLoaderClient.h:
* Source/WebCore/loader/NavigationDisabler.h:
* Source/WebCore/loader/NavigationScheduler.cpp:
* Source/WebCore/loader/NavigationScheduler.h:
* Source/WebCore/loader/PolicyChecker.h:
* Source/WebCore/loader/ProgressTracker.cpp:
* Source/WebCore/loader/ProgressTracker.h:
* Source/WebCore/loader/ProgressTrackerClient.h:
* Source/WebCore/loader/ResourceLoadObserver.h:
* Source/WebCore/loader/ResourceTiming.h:
* Source/WebCore/loader/SubframeLoader.h:
* Source/WebCore/loader/SubresourceLoader.h:
* Source/WebCore/loader/TextTrackLoader.h:
* Source/WebCore/loader/ThreadableLoaderClient.h:
* Source/WebCore/loader/WorkerThreadableLoader.h:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.cpp:
* Source/WebCore/loader/appcache/ApplicationCacheGroup.h:
* Source/WebCore/loader/appcache/ApplicationCacheHost.h:
* Source/WebCore/loader/archive/ArchiveResourceCollection.h:
* Source/WebCore/loader/cache/CachedFontLoadRequest.h:
* Source/WebCore/loader/cache/CachedResource.h:
* Source/WebCore/loader/cache/CachedResourceLoader.h:
* Source/WebCore/loader/cache/CachedResourceRequestInitiatorTypes.h:
* Source/WebCore/loader/cache/CachedSVGDocumentReference.h:
* Source/WebCore/loader/cache/MemoryCache.h:
* Source/WebCore/loader/icon/IconLoader.h:
* Source/WebCore/loader/ios/LegacyPreviewLoader.h:

Canonical link: https://commits.webkit.org/267205@main
